dali/sasha/sadbghk.cpp

```diff
@@ -0,0 +1,180 @@
+#include "platform.h"
+
+#include "jfile.hpp"
+#include "jlib.hpp"
+#include "jlog.hpp"
+#include "jiface.hpp"
+#include "jptree.hpp"
+#include "jstring.hpp"
+
+#include "saserver.hpp"
+#include "sautil.hpp"
+
+static constexpr unsigned defaultExpiryDays = 14;
+static constexpr unsigned defaultHouseKeepingIntervalHours = 24;
+
+#define LOGDBGHK "DEBUGPLANEHOUSEKEEPING: "
+
+// Debug Housekeeping monitor
+
+class CSashaDebugPlaneHousekeepingServer : public ISashaServer, public Thread
+{
+    std::atomic<bool> stopped{false};
+    Semaphore stopSem;
+    Mutex runMutex;
+    StringBuffer debugDir;
+    unsigned expiryDays{0};
+
+public:
+    IMPLEMENT_IINTERFACE_USING(Thread);
+
+    CSashaDebugPlaneHousekeepingServer()
+        : Thread("CSashaDebugPlaneHousekeepingServer")
+    {
+    }
+
+    virtual void start() override
+    {
+        expiryDays = getComponentConfigSP()->getPropInt("@expiryDays", defaultExpiryDays);
+        if (expiryDays == 0)
+        {
+            WARNLOG("Failed to get expiry days for debug plane");
```

Comment:
Not "failed to get" - it had to be deliberately configured as 0 for this to be true.
I would treat as disabled and exit without starting.

In fact a better pattern is to set 'stopped' to true in member initialization,
and set to false just before the Thread::start() call.

It would also be good to check 'interval' here for 0, and not start, rather than check it once started and then stop.